src/Middleware/RequestDecompression/src/SizeLimitedStream.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.RequestDecompression;
+
+internal sealed class SizeLimitedStream : Stream
+{
+    private readonly Stream _innerStream;
+    private readonly long? _sizeLimit;
+
+    private long _totalBytesRead;
+
+    public SizeLimitedStream(Stream innerStream, long? sizeLimit)
+    {
+        if (innerStream is null)
+        {
+            throw new ArgumentNullException(nameof(innerStream));
+        }
+
+        _innerStream = innerStream;
+        _sizeLimit = sizeLimit;
+    }
+
+    public override bool CanRead => _innerStream.CanRead;
+
+    public override bool CanSeek => _innerStream.CanSeek;
+
+    public override bool CanWrite => false;
```

Comment:
pass all the write operation to the inner stream, even if we don't expect them to be supported. Blocking write's is not this stream's goal.

Reply:
Good call. I've updated `CanWrite`, `Write`, and `Flush` to pass the operations to the inner stream.